cloudformation.yaml

```
@@ -147,6 +148,36 @@ Resources:
                   - autoscaling:DescribeAutoScalingGroups
                   - autoscaling:DescribeAutoScalingInstances
                 Resource: '*'
+        - PolicyName: SendEmailSES
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action: ses:SendEmail
+                Resource: '*'
+  GithubActionsSESSendEmailsRole:
+    Type: AWS::IAM::Role
+    Condition: NotIsProd
+    Properties:
+      RoleName: GithubActionsSESSendEmailsRole
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRoleWithWebIdentity
+            Principal:
+              Federated: arn:aws:iam::942464564246:oidc-provider/token.actions.githubusercontent.com
```

should this ARN be public?

---

I believe ARNs can be public but we can make it a GitHub secret if we prefer?

---

this is in cloudformation, so it can't go in github secrets, i'm just saying this as other ARNs we use in this file are pulled in as parameters rather than made public

---

e.g. https://github.com/guardian/gateway/blob/19f03a7ba11810b2bc88540871dbeb447c30fa35/cloudformation.yaml#L29

---

and then you add the value of the parameter in cloudformation console

---

generally ARNs aren't made public as it may give away information about the infrastructure of our AWS account

---

actually in this case you might just want to hide the account id

similar to what we do elsewhere in the file e.g. https://github.com/guardian/gateway/blob/19f03a7ba11810b2bc88540871dbeb447c30fa35/cloudformation.yaml#L134

```suggestion
              Federated: arn:aws:iam::${AWS::AccountId}:oidc-provider/token.actions.githubusercontent.com
```